telethon/telegram_client.py

```diff
@@ -476,6 +476,55 @@ def get_message_history(self,
 
         return total_messages, result.messages, entities
 
+    def get_all_participants(self,entity):
+        """
+        Returns all the partcipants for the specified entity.
+
+        :param entity:      The entity from whom to retrieve the participants
+
+        :return: A list of participants for the given entity
+        """
+
+        def update_participant_set(participants, participant):
+            """
+            Updates the set of participants for the specified entity.
+
+            :param participants:    Set of participants in the chat or group
+            :param participant:     Participant in the chat or group.
+
+            :return: void
+            """
+            if participant in participants:
+                participants.remove(participant)
+            else:
+                participants.add(participant)
+
+        entity_type = self.get_entity(entity)
+        if isinstance(entity_type,User):
+            return [entity]
+        elif isinstance(entity_type,Chat):
+
+            participants = set()
+            _ ,messages, _ = self.get_message_history(entity)
```

Comment:
From my understanding there does not seem to be a way to directly get the current members in a chat through the API, so you have to keep track of the members yourself, at least from what it seems. But, I'll still see if there is another way.

Reply:
also this is not getting all the users who have ever participated, it is going through the 'action' messages to update the participants set. So in reality it returns the current member list just as long as you process all the 'action' messages

Reply:
> So in reality it returns the current member list just as long as you process all the 'action' messages

Including actions of kicked, joined, left, invitted… in the whole chat history. There must be a better request, how else could Telegram Desktop and such load it so fast otherwise?

Reply:
I'm pretty sure the app only show up to 200 members, which are also recent members. There is an API call to get the recent members but not to get the entire list.

Reply:
I don't have groups so big to test that with them but ideally please check how official applications do it so we can at least mimic that if you can't find a solution.